[docs][PLAT-6522] Docs for YBA Kubernetes Multi-Cluster Services (#16536)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md

   Note down the cluster names from the cluster set, these will be used during the cloud provider setup in YugabyteDB Anywhere.

### Prepare Kubernetes clusters for Istio mulit-cluster

Istio service mesh can span across multiple clusters which allows us to configure MCS. It supports different topologies and network configurations. Follow [Install Multicluster](https://istio.io/latest/docs/setup/install/multicluster/) to get started with the multi-cluster Istio setup.

The Istio configuration for each cluster should have following options:

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    defaultConfig:
      proxyMetadata:
        ISTIO_META_DNS_CAPTURE: "true"
        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
  # rest of the configuration…
```
Refer the Istio setup section from [Multi-Region YugabyteDB Deployments on Kubernetes with Istio](https://www.yugabyte.com/blog/multi-region-yugabytedb-deployments-on-kubernetes-with-istio/) for a more step by step guide along with explanation of the options being used.

### Configure the cloud provider for MCS

> Once you have the cluster setup ready, you can follow [Configure the Kubernetes cloud provider](#). And refer to this section for region and zone configuration requried for multi-region universes.

Reviewer: Link to come?

Author: Want it to jump to the top of the page in case someone has directly reading the MCS sections, hence the `#`.

#### Configure region and zone for GKE MCS
Set the following values for all the zones from your Kubernetes clusters connected via GKE MCS.

> 1. Specify all other fields like Region, Zone, etc. by following [Configure region and zones](#configure-region-and-zones).

Reviewer: ```suggestion
1. Specify fields such as Region, Zone, and so on as you would normally.
```

1. Set the **Cluster DNS Domain** to `clusterset.local`.
1. Upload the correct **Kube Config** of the cluster.
1. Set the **Pod Address Template** to `{pod_name}.<cluster membership name>.{service_name}.{namespace}.svc.{cluster_domain}`, where the `<cluster membership name>` is the membership name of the Kubernetes cluster set during the fleet setup.
1. Set the **Overrides** as follows:
   ```yaml
   multicluster:
     createServiceExports: true
     kubernetesClusterId: "<cluster membership name>"
     mcsApiVersion: "net.gke.io/v1"
   ```
For example, if your cluster membership name is `yb-asia-south1`, then the **Add new region** screen would look as follows:

> ![Add new region screen of YugabyteDB Anywhere with GKE MCS](/images/ee/k8s-setup/k8s-add-region-gke-mcs.png)

Reviewer: Is the Add region screen in the earlier section out of date?

Author: Yes, but in latest version new fields have been added. Should I update that screenshot as part of this PR?

Author: I will create a ticket if required. The current screenshot we have in earlier section is rather simple and sufficient. Right now we don't talk about the other options here in this doc as well.

#### Configure region and zones for OpenShift MCS

> Follow [Configure the OpenShift cloud provider](../openshift/) and set the following values for all the zones from your Kubernetes clusters connected via OpenShift MCS (Submariner).

Reviewer: ```suggestion
Follow the instructions in [Configure the OpenShift cloud provider](../openshift/) and [Create a provider in YugabyteDB Anywhere](../openshift/#create-a-provider-in-yugabytedb-anywhere). For all the zones from your Kubernetes clusters connected via OpenShift MCS (Submariner), add a region as follows:
```

> 1. Specify all other fields like Region, Zone, etc. by following [Create a provider in YugabyteDB Anywhere](../openshift/#create-a-provider-in-yugabytedb-anywhere).

Reviewer: ```suggestion
Specify fields such as Region, Zone, and so on as you would normally.
```

1. Set the **Cluster DNS Domain** to `clusterset.local`.
1. Upload the correct **Kube Config** of the cluster.
1. Set the **Pod Address Template** to `{pod_name}.<cluster name>.{service_name}.{namespace}.svc.{cluster_domain}`, where the `<cluster name>` is the name of the OpenShift cluster set during the cluster set creation.
1. Set the **Overrides** as follows:
   ```yaml
   multicluster:
     createServiceExports: true
     kubernetesClusterId: "<cluster name>"
   ```
For example, if your cluster name is `yb-asia-south1`, then the values will be as follows:
- **Pod Address Template**  
  `{pod_name}.yb-asia-south1.{service_name}.{namespace}.svc.{cluster_domain}`
- **Overrides**
  ```yaml
  multicluster:
    createServiceExports: true
    kubernetesClusterId: "yb-asia-south1"
  ```

#### Configure region and zones for Istio

> Set the following values for all the zones from your Kubernetes clusters connected via Istio.

Reviewer: ```suggestion
Follow the steps in [Configure region and zones](#configure-region-and-zones) and set values for all the zones from your Kubernetes clusters connected via Istio as follows.
```

> 1. Specify all other fields like Region, Zone, etc. by following [Configure region and zones](#configure-region-and-zones).

Reviewer: ```suggestion
1. Specify fields such as Region, Zone, and so on as you would normally.
```

  For more information, see [Helm chart: Prerequisites](../../../../deploy/kubernetes/single-zone/oss/helm-chart/#prerequisites).

- Overrides to use a secret for LDAP authentication. Refer to [Create secrets for Kubernetes](../../../../secure/authentication/ldap-authentication-ysql/#create-secrets-for-kubernetes).

> ## Configure Multi-Cluster Services

Reviewer 2: Configure the provider for multi-cluster environments?

Author: Made it "Configure Kubernetes multi-cluster environment", because using "provider" becomes confusing, as we have one more section with this title: "Configure the cloud provider for MCS"

If you plan to create multi-region YugabyteDB universes, you can setup [Multi-Cluster Services](https://git.k8s.io/enhancements/keps/sig-multicluster/1645-multi-cluster-services-api) (MCS) across your Kubernetes clusters. This section covers implementation specific details for setting up MCS on various cloud providers and service mesh tools.

{{< warning title="MCS support is in beta" >}}

MCS support in YugabyteDB Anywhere is currently in the beta state. It comes with following caveats:
- The metrics screen of universe might not show correct metrics for all the pods.

> - xCluster replication needs manual intervention to work on OpenShift MCS.

Reviewer 2: an additional manual step to work on ...

> - There can be some untested combinations and edge cases which might have bugs.

Reviewer 2: Let's remove the untested part of it because it sounds too scary. The "edge cases" part is implied by the beta label.

Author: Yes, I was a bit unsure about it. Removed it.

> The Kubernetes Multi-Cluster Services API is currently in alpha state, though there are various implementation of MCS which are [considered to be stable](https://github.com/kubernetes-sigs/mcs-api/issues/17#issuecomment-1309073682). To know more, see [API versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning).

Reviewer 2: let's move the "k8s api is currently in alpha state" to right after the warning.

Author: Done, check if it is clear enough.

{{< /warning >}}

> ### Configure GKE Multi-Cluster Services

Reviewer 2: should we call these three sections 

### Prepare your kubernetes cluster for GKE MCS

   Note down the cluster names from the cluster set, these will be used during the cloud provider setup in YugabyteDB Anywhere.

### Configure Istio for multi-region

Istio service mesh can span across multiple clusters which allows us to configure MCS. It supports different topologies and network configurations. Follow [Install Multicluster](https://istio.io/latest/docs/setup/install/multicluster/) to get started with the multi-cluster Istio setup.

The Istio configuration for each cluster should have following options:

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    defaultConfig:
      proxyMetadata:
        ISTIO_META_DNS_CAPTURE: "true"
        ISTIO_META_DNS_AUTO_ALLOCATE: "true"
  # rest of the configuration…
```
Refer the Istio setup section from [Multi-Region YugabyteDB Deployments on Kubernetes with Istio](https://www.yugabyte.com/blog/multi-region-yugabytedb-deployments-on-kubernetes-with-istio/) for a more step by step guide along with explanation of the options being used.

> TODO: the explanation about these two proxyMetadata options is in the above blog post. Should we add that here itself? It can also be set with an annotation on pod.

Reviewer 2: the level of detail we have right now is sufficient.

> ### Create provider

Reviewer 2: and we can call this

Configure the cloud provider for MCS
